### PR TITLE
Entry impromevents and bug fixes

### LINKED ISSUE #313: Submitting an empty url crashes the server
# Bug report

## Expected Behavior
When submitting an entry without a optional url, the program should still save the entry.

## Actual Behavior
The sever crashes with the following error:
image

## Steps to Reproduce the Problem
1. Create a template where the student can submit an entry with a url
2. As a student submit an entry without the url

### LINKED ISSUE #321: Editing format removes already submitted files
# Bug report

## Expected Behavior
When editing a format, it should just keep the current submitted file.

## Actual Behavior
It removes the files

## Steps to Reproduce the Problem
1. go to an entry where a file is uploaded
2. click onedit
3. click on save

### LINKED ISSUE #322: The ability to remove your own entries
# Feature request

## Description
The ability to remove your own entries. This maybe should only be available before it is graded

## User story
Go to your entry -> click on delete -> confirm deletion -> get redirected back to the journal, but now without the deleted entry

### PR DESCRIPTION
**Pull request by Larspolo**
## Description
Added the ability to remove entries and fixed entry submission issues.

## Summary of changes
- Fixes #322 
- Fixes #313 
- Fixes #321

### CODE REVIEW

**Comment by Larspolo** [comment]
@mjvkeulen the rich text editor werkt nog niet in de edit mode, kan je daarnaar kijken?

File: src/vue/src/components/entry/EntryNode.vue [hunk 1]
```diff
@@ -220,6 +229,7 @@ export default {
 
                 for (var content of this.entryNode.entry.content) {
                     if (content.field === templateField.id) {
+                        console.log(content)
```
Comment (engelhamer) [inline src/vue/src/components/entry/EntryNode.vue]: Left a debug log.

Comment (Larspolo) [reply src/vue/src/components/entry/EntryNode.vue]: thx

File: src/django/VLE/views/entry.py [hunk 2]
```diff
@@ -174,19 +181,56 @@ def partial_update(self, request, *args, **kwargs):
...
+            success -- with a message that the course was deleted
+        """
+        if not request.user.is_authenticated:
+            return response.unauthorized()
+        pk = kwargs.get('pk')
+
+        try:
+            entry = Entry.objects.get(pk=pk)
+        except Entry.DoesNotExist:
+            return response.not_found('Course does not exist.')
```
Comment (mjvkeulen) [inline src/django/VLE/views/entry.py]: Should this not be entry does not exists?

File: src/django/VLE/views/entry.py [hunk 3]
```diff
@@ -174,19 +181,56 @@ def partial_update(self, request, *args, **kwargs):
...
+
+        if not permissions.has_assignment_permission(request.user, entry.node.journal.assignment, 'can_edit_journal'):
+            return response.forbidden('You can only delete your own entries.')
+
+        if entry.grade:
+            return response.forbidden('This anty is already graded, and thus cannot be deleted anymore.')
```
Comment (mjvkeulen) [inline src/django/VLE/views/entry.py]: anty -> entry. 

You cannot delete graded entries.

**Pull request merged**